segmentation/scripts/instance_segm.py

```diff
@@ -0,0 +1,281 @@
+    Args:
+        predictions (list): List of predictions, where each element contains
+                            [initial_image, true_mask, predicted_mask].
+        target_rgb (tuple): RGB color value representing the class to segment (default is (255, 255, 255)).
+
+    Returns:
+        results (list): List of dictionaries with keys 'image', 'true_instance_mask', and 'predicted_instance_mask'.
+    """
+    results = []
+    for prediction in predictions:
+        initial_image = prediction[0]
+        true_mask, _ = semantic_to_instance_mask(prediction[1],
+                                                 target_rgb=target_rgb)
+        predicted_mask, _ = semantic_to_instance_mask(prediction[2],
+                                                      target_rgb=target_rgb)
+
+        results.append({
+            "image": initial_image,
+            "true_instance_mask": true_mask,
+            "predicted_instance_mask": predicted_mask,
+        })
+
+    return results
+
+
+def visualize_instance_results(results):
+    """
+    Visualize all results with initial image, true instance mask, and predicted instance mask.
+
+    Args:
+        results (list): List of dictionaries with keys 'image', 'true_instance_mask', and 'predicted_instance_mask'.
+    """
+    for result in results:
+        image = result["image"]
+        true_mask_rgb = visualize_instance_mask(result["true_instance_mask"])
+        predicted_mask_rgb = visualize_instance_mask(
+            result["predicted_instance_mask"])
+
+        # Plot the results
+        plt.figure(figsize=(15, 5))
+
+        # Initial image
+        plt.subplot(1, 3, 1)
+        plt.title("Initial Image")
+        plt.imshow(image)
+        plt.axis("off")
+
+        # True instance mask
+        plt.subplot(1, 3, 2)
+        plt.title("True Instance Mask")
+        plt.imshow(true_mask_rgb)
+        plt.axis("off")
+
+        # Predicted instance mask
+        plt.subplot(1, 3, 3)
+        plt.title("Predicted Instance Mask")
+        plt.imshow(predicted_mask_rgb)
+        plt.axis("off")
+
+        plt.show()
+
+
+def iou(true_mask, pred_mask):
+    """
+    Calculate Intersection over Union (IoU) between two binary masks.
+
+    Args:
+        true_mask (numpy.ndarray): True binary mask.
+        pred_mask (numpy.ndarray): Predicted binary mask.
+
+    Returns:
+        float: IoU value between 0 and 1.
+    """
+    intersection = np.logical_and(true_mask, pred_mask)
+    union = np.logical_or(true_mask, pred_mask)
+    return np.sum(intersection) / np.sum(union) if np.sum(union) != 0 else 0.0
+
+
+def get_contour_bounding_boxes(contours):
+    """
+    Get bounding boxes for a list of contours.
+
+    Args:
+        contours (list): List of contours from cv2.findContours.
+
+    Returns:
+        list: List of bounding boxes [(x, y, w, h)].
+    """
+    return [cv2.boundingRect(contour) for contour in contours]
+
+
+def filter_contours_by_bbox(true_contours, pred_contours, true_bboxes,
```

Comment:
Might also be interesting idea to try and merge the intersecting predictions or ones that are within some predefined proximity to each other.